frontend/hub/collections/hooks/useCopyToRepository.tsx

```diff
@@ -0,0 +1,252 @@
+import { usePageDialog } from './../../../../framework';
+import { Button, Modal, ModalVariant } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect, useCallback } from 'react';
+import {
+  useRepositoryColumns,
+  useRepositoryFilters,
+} from './../../repositories/hooks/useRepositorySelector';
+import { PageTable } from './../../../../framework/PageTable/PageTable';
+import { usePulpView } from '../../usePulpView';
+import { AnsibleAnsibleRepositoryResponse } from './../../api-schemas/generated/AnsibleAnsibleRepositoryResponse';
+import { pulpAPI, hubAPI, hubAPIPost } from '../../api/utils';
+import { useGetRequest } from './../../../common/crud/useGet';
+import { HubItemsResponse } from '../../useHubView';
+import { PulpItemsResponse } from '../../usePulpView';
+import { parsePulpIDFromURL } from '../../api/utils';
+import { useHubContext, HubContext } from './../../useHubContext';
+import { SigningServiceResponse } from '../../api-schemas/generated/SigningServiceResponse';
+import { AwxError } from '../../../awx/common/AwxError';
+
+export function useCopyToRepository() {
+  const [_, setDialog] = usePageDialog();
+
+  const context = useHubContext();
+  const onClose = useCallback(() => setDialog(undefined), [setDialog]);
+
+  return (collection: CollectionVersionSearch, operation: 'approve' | 'copy') => {
+    setDialog(
+      <CopyToRepositoryModal
+        collection={collection}
+        context={context}
+        onClose={onClose}
+        operation={operation}
+      />
+    );
+  };
+}
+
+function CopyToRepositoryModal(props: {
+  collection: CollectionVersionSearch;
+  context: HubContext;
+  onClose: () => void;
+  operation: 'approve' | 'copy';
+}) {
+  const toolbarFilters = useRepositoryFilters();
+  const tableColumns = useRepositoryColumns();
+  const { t } = useTranslation();
+  const { collection } = props;
+  const request = useGetRequest<HubItemsResponse<CollectionVersionSearch>>();
+  const pulpRequest = useGetRequest<PulpItemsResponse<SigningServiceResponse>>();
+
+  const [selectedRepositories, setSelectedRepositories] = useState<Repository[]>([]);
+  const [fixedRepositories, setFixedRepositories] = useState<Repository[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string>('');
+
+  const { collection_auto_sign, require_upload_signatures } = props.context.featureFlags;
+  const autoSign = collection_auto_sign && !require_upload_signatures;
+
+  const operation = props.operation;
+
+  const copyToRepositories = async () => {
+    try {
+      setIsLoading(true);
+
+      const { repository } = props.collection;
+      if (!repository) {
+        return;
+      }
+
+      const pulpId = parsePulpIDFromURL(repository?.pulp_href);
+
+      let signingService = '';
+
+      if (autoSign) {
+        const signingServiceName = props.context.settings.GALAXY_COLLECTION_SIGNING_SERVICE;
```

Comment:
`autoSign` is only relevant for approval, not for copy
for copy, the important thing is whether a signing service is configured

```suggestion
      const signingServiceName = props.context.settings.GALAXY_COLLECTION_SIGNING_SERVICE;
      if (((operation === 'approve' && autoSign) || (operation === 'move')) && signingServiceName) {
```

Reply:
Thanks, signing is still mystery for me.